plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -0,0 +1,116 @@
+package com.google.cloud.tools.eclipse.usagetracker;
+
+import com.google.api.client.util.escape.PercentEscaper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import org.eclipse.core.runtime.Platform;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Provides methods that report plugin-specific events to Analytics.
+ */
+public class AnalyticsPingManager {
+
+  private static final String ANALYTICS_COLLECTION_URL = "https://ssl.google-analytics.com/collect";
+  // TODO: retrieve tracking ID from a system environment variable.
+  private static final String ANALYTICS_TRACKING_ID = "@ga.tracking.id@";
+  private static final String APPLICATION_NAME = "gcloud-eclipse-tools";
+
+  // TODO: see if there a way to get a unique ID for each Eclipse installation.
```

Comment:
If this UUID initialization seems ok to you, I think it's fine. I asked this because our IntelliJ plugin has an API `getInstallationUUID()`, and we have been using it actually.